### PR TITLE
A template for a Journal of Statistical Software article

### PR DESCRIPTION
**Pull request by RLesur**
Here's a proposal for the JSS template.  

Before explaining the main characteristics of this template, I have to mention a modification of the `html_paged()` output format I propose in this PR.

### Modification introduced in `html_paged()`
The proposition is to introduce a new `csl` argument for passing a _Citation Style Language_ (CSL) file to `html_paged()`.   
I needed such a file for the JSS template and realized that this argument could be useful for other templates.

#### Screenshot and demo file

Here's a screenshot of RStudio...
JSS

... and a PDF file: [JSS_demo.pdf](https://github.com/rstudio/pagedown/files/2747567/HTML.and.CSS.Paged.Media.for.Scientific.Publications.with.the.pagedown.R.Package.pdf).

#### Main characteristics of the PR
- **markdown first**: as you can see in `inst/rmarkdown/templates/jss-paged/skeleton/skeleton.Rmd`, the user is encouraged to use markdown but I also added support for LaTeX syntax.
- **fonts**: all the fonts are downloaded from the RStudio MathJax CDN. There's no font in this PR: I did not include the _TeX Gyre Chorus_ font for the JSS title. The template use the following image provided by the JSS website (https://www.jstatsoft.org/public/journals/1/homeHeaderTitleImage_en_US.png)  

If you see nothing, it is normal. There's a CSS trick to make it visible ;-)
- **CSL file**: I think that the only mean we have to deal with citations styling in **pagedown** is to use a CSL file. The problem I faced is that the JSS citations style is not referenced in the Zotero Style Repository. So, I've built such a file. I haven't spent too much time on this task: it is a "quick and dirty" CSL file. I hope that the JSS editors will develop an official CSL file one day.
- **small bugfix** in `inst/resources/js/config.js` for custom templates that do not possess a front matter.

I also needed to use some hacks for difficult tasks. I let you discover them by reviewing the PR :smiley: .

### CODE REVIEW

File: inst/resources/css/jss.css [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  background-clip: border-box;
+  background-image: url(https://www.jstatsoft.org/public/site/images/jstatsoft/jsslogo.jpg);
+  background-size: contain;
+  background-repeat: no-repeat;
+  background-origin: border-box;
+}
+.journal-name {
+  padding-top: 14px;
+  padding-left: 20px;
+  filter: invert(100%);
```
Comment (yihui) [inline inst/resources/css/jss.css]: I see :)

**Comment by RLesur** [comment]
Don't hesitate to ask me if you have any question. I was so hesitant about some decisions. I often wonder what would be the best weapon (HTML, CSS, JS, pandoc variables or filters...) for a specific task

For instance, I struggled a lot with the author blocks layout. I've implemented the rank computation in the pandoc filter while the same computation could be done in JS.
In many cases, I hesitate to implement some features in pandoc filters or JS. Tell me if you have a preferred option.

I also decided to use a specific pandoc template: it was much more easy. But maybe the whole work could be done with the html_paged default template and a bunch of JS scripts. It seems to me more rational to use markup instead of JS in this case. But maybe I'm wrong.

File: inst/resources/css/jss.css (hunk 1 above)
Comment (RLesur) [reply inst/resources/css/jss.css]: this is so weird (it was funny to do but I definitely don't like that).

**Comment by yihui** [comment]
I think I have given ourselves enough excuse in my talk for screwing things up by telling them this package is yough and experimental, so feel free to choose a path that feels most convenient to you. If I see something worth discussing, I'll surely let you know.

I'm poor at Pandoc filters, so I often resort to JS hacks. I don't really care much about the implementation. As long as we are able to "[fold the blinds](https://media.giphy.com/media/yYSSBtDgbbRzq/giphy.gif)", I don't mind how exactly we pull the strings. 😁

**Pull request merged**